charms/katib-controller/tests/integration/test_charm.py

```diff
@@ -46,6 +39,80 @@
     % {"katib_version": KATIB_VERSION},
 }
 
+CUSTOM_IMAGES_PATH = (
+    Path(__file__).resolve().parent.parent.parent / "src" / "default-custom-images.json"
+)
+with CUSTOM_IMAGES_PATH.open() as f:
+    CUSTOM_IMAGES = json.load(f)
+
+EXPECTED_KATIB_CONFIG = {
+    "katib-config.yaml": f"""---
+apiVersion: config.kubeflow.org/v1beta1
+kind: KatibConfig
+init:
+  controller:
+    webhookPort: 443
+    trialResources:
+      - Job.v1.batch
+      - TFJob.v1.kubeflow.org
+      - PyTorchJob.v1.kubeflow.org
+      - MPIJob.v1.kubeflow.org
+      - XGBoostJob.v1.kubeflow.org
+runtime:
+  metricsCollectors:
+    - kind: StdOut
+      image: {CUSTOM_IMAGES["metrics_collector_sidecar__stdout"]}
+    - kind: File
+      image: {CUSTOM_IMAGES["metrics_collector_sidecar__file"]}
+    - kind: TensorFlowEvent
+      image: {CUSTOM_IMAGES["metrics_collector_sidecar__tensorflow_event"]}
+      resources:
+        limits:
+          memory: 1Gi
+  suggestions:
+    - algorithmName: random
+      image: {CUSTOM_IMAGES["suggestion__random"]}
+    - algorithmName: tpe
+      image: {CUSTOM_IMAGES["suggestion__tpe"]}
+    - algorithmName: grid
+      image: {CUSTOM_IMAGES["suggestion__grid"]}
+    - algorithmName: hyperband
+      image: {CUSTOM_IMAGES["suggestion__hyperband"]}
+    - algorithmName: bayesianoptimization
+      image: {CUSTOM_IMAGES["suggestion__bayesianoptimization"]}
+    - algorithmName: cmaes
+      image: {CUSTOM_IMAGES["suggestion__cmaes"]}
+    - algorithmName: sobol
+      image: {CUSTOM_IMAGES["suggestion__sobol"]}
+    - algorithmName: multivariate-tpe
+      image: {CUSTOM_IMAGES["suggestion__multivariate_tpe"]}
+    - algorithmName: enas
+      image: {CUSTOM_IMAGES["suggestion__enas"]}
+      resources:
+        limits:
+          memory: 400Mi
+    - algorithmName: darts
+      image: {CUSTOM_IMAGES["suggestion__darts"]}
+    - algorithmName: pbt
+      image: {CUSTOM_IMAGES["suggestion__pbt"]}
+      persistentVolumeClaimSpec:
+        accessModes:
+          - ReadWriteMany
+        resources:
+          requests:
+            storage: 5Gi
+  earlyStoppings:
+    - algorithmName: medianstop
+      image: {CUSTOM_IMAGES["early_stopping__medianstop"]}
+"""
+}
```

Comment:
this is already an improvement, a suggestion is to get the katib config from the template in https://github.com/canonical/katib-operators/blob/main/charms/katib-controller/src/templates/katib-config.yaml.j2 and render it with the `CUSTOM_IMAGES`. This way on upgrades we do not have to change the test file, let's say they add a new algorithm, job type, or so.